orttraining/orttraining/python/orttraining_pybind_state.cc

```diff
@@ -823,18 +823,28 @@ void addObjectMethodsForTraining(py::module& m, ExecutionProviderRegistrationFn
           GradientDefinitionRegistry::Instance().SetStopGradientEdgesForNode(key, edges);
         });
 #ifdef ENABLE_TRAINING_ON_DEVICE
-  // Python apis only supports CPU device for now.
-  // TODO(adamlouly) : Add support for CUDA device.
   py::class_<onnxruntime::training::api::Module> training_module(m, "Module", R"pbdoc(Training Module.)pbdoc");
   training_module
       .def(py::init([](const std::string& model_uri,
                        onnxruntime::training::api::CheckpointState& state,
-                       std::optional<std::string> eval_model_uri) {
+                       std::optional<std::string> eval_model_uri,
+                       std::optional<std::string> provider_type) {
```

Comment:
Considering that these are always sent in from the python frontend, shall we remove the `std::optional`?

Reply:
eval_model_uri is optional you can pass it or not, also provider_type can be option, if its not passed from the front end, it will use IExecutionProvider to initialize the Module